From a992be100f5acdb96811d5c40e0ff4980c4324a7 Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Mon, 31 Aug 2026 11:30:30 -0300
Subject: [PATCH] dt-bindings: net: airoha: extend WED v1 SoC support

Describe the EN751221 and EN7528 compatibles alongside EN7523. The EN7528 variant also requires an SCU phandle for the vendor LAN arbitration setting.

Signed-off-by: Codex <codex@openai.com>
Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 .../bindings/net/airoha,en7523-wed.yaml       | 26 ++++++++++++++++---
 1 file changed, 22 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/airoha,en7523-wed.yaml b/Documentation/devicetree/bindings/net/airoha,en7523-wed.yaml
index 35ba702c3..157655bf0 100644
--- a/Documentation/devicetree/bindings/net/airoha,en7523-wed.yaml
+++ b/Documentation/devicetree/bindings/net/airoha,en7523-wed.yaml
@@ -4,7 +4,7 @@
 $id: http://devicetree.org/schemas/net/airoha,en7523-wed.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: Airoha EN7523 Wireless Ethernet Dispatch controller
+title: Airoha and EcoNet WED v1 controller
 
 maintainers:
   - Lorenzo Bianconi <lorenzo@kernel.org>
@@ -12,15 +12,23 @@ maintainers:
 description:
   The Wireless Ethernet Dispatch (WED) controller connects a PCIe wireless
   DMA engine to the Ethernet frame engine and offloads the wireless transmit
-  datapath. Each EN7523 WED instance is associated with one PCIe domain and
-  one WDMA instance.
+  datapath. Each WED instance is associated with one PCIe domain and one
+  WDMA instance.
 
 properties:
   compatible:
     items:
-      - const: airoha,en7523-wed
+      - enum:
+          - econet,en751221-wed
+          - econet,en7528-wed
+          - airoha,en7523-wed
       - const: syscon
 
+  airoha,scu:
+    $ref: /schemas/types.yaml#/definitions/phandle
+    description:
+      System controller used to select LAN arbitration mode on EN7528.
+
   reg:
     items:
       - description: WED register block
@@ -40,6 +48,16 @@ required:
   - reg-names
   - interrupts
 
+allOf:
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: econet,en7528-wed
+    then:
+      required:
+        - airoha,scu
+
 additionalProperties: false
 
 examples:
-- 
2.55.0

